events/entity_counts.go

```diff
@@ -0,0 +1,97 @@
+package events
+
+import (
+	"fmt"
+	"github.com/openziti/edge/controller/persistence"
+	"github.com/openziti/foundation/util/cowslice"
+	"github.com/pkg/errors"
+	"go.etcd.io/bbolt"
+	"reflect"
+	"time"
+)
+
+const EntityCountEventNS = "edge.entityCounts"
+
+type EntityCountEvent struct {
+	Namespace string           `json:"namespace"`
+	Timestamp time.Time        `json:"timestamp"`
+	Counts    map[string]int64 `json:"counts"`
+	Error     string           `json:"error"`
+}
+
+func (event *EntityCountEvent) String() string {
+	return fmt.Sprintf("%v timestamp=%v counts=%+v err=%v",
+		event.Namespace, event.Timestamp, event.Counts, event.Error)
+}
+
+var entityCountEventGenerator func(interval time.Duration)
```

Comment:
I only see this declared here and set once in `Initi(...)`. Is this for tests?

Reply:
The only other reason I can see is so you can use the closure scope for the closeNotify chan.

/shrug

Reply:
It's used in registerEntityCountEventHandler

Reply:
It's just a place to store the state until register gets called.

Reply:
I didn't see why you would use a module-level `var` declared function over a regularly defined function other than the close notify closure scope.

It would be just as easy to use the close notify as a package-level variable. The only reason I've really seen `var` declared module functions is to monkey patch during testing.

Reply:
it's not just the close notify, it's also holding on to the the provider and stores. I could either store them all in individual vars, or a struct, or a func, which has the benefit of being a single var and opaque.

Reply:
Thanks, no change is needed. 

Interesting pattern. I would have encapsulated that data into another struct that had its own `Run()` or like function. Would do the same to facilitate scope and testing w/o using module-level overrides.